ucrop/src/main/java/com/yalantis/ucrop/task/BitmapCropTask.java

```diff
@@ -204,7 +204,7 @@ private void copyExifForOutputFile(Context context) throws IOException {
         }
     }
 
-    private void saveImage(@NonNull Bitmap croppedBitmap) {
+    private void saveImage(@NonNull Bitmap croppedBitmap) throws IOException {
```

Comment:
Your PR description does not match the changes you have made, would you like to elaborate on the issue?

Reply:
When introduced to a nonexistent path, will throw FileNotFoundException, at this point in the `onPostExecute` method, should the callback ` mCropCallback#OnCropFailure ` method, But the actual callback `mCropCallback#onBitmapCropped` method

Reply:
@liujingxing  is this code working properly? could you please add a crop from camera feature?

Reply:
```java
public class TestActivity extends AppCompatActivity {

    private void crop(File srcFile) {
        String cropPath = getExternalCacheDir() + "/image/test.jpg";
        File cropFile = new File(cropPath);
        File parentFile = cropFile.getParentFile();
        //is false，The BitmapCropTask#saveImage method will throw a FileNotFoundException
        boolean exists = parentFile.exists(); 
        UCrop.of(Uri.fromFile(srcFile), Uri.fromFile(cropFile))
            .withAspectRatio(1, 1)
            .withMaxResultSize(800, 800)
            .start(this);
    }

    @Override
    protected void onActivityResult(int requestCode, int resultCode, @Nullable Intent data) {
        super.onActivityResult(requestCode, resultCode, data);
        if (resultCode == RESULT_OK) {
            if (requestCode == UCrop.REQUEST_CROP) {
                Uri uri = UCrop.getOutput(data);
                File cropFile = new File(uri.getPath());
                boolean exists = cropFile.exists(); // Actual false, expected true
            }
        } else if (resultCode == UCrop.RESULT_ERROR) {
            //FileNotFoundException occurred and should be called back here
            Throwable throwable = UCrop.getError(data);
        }
    }
}
```

Reply:
Could you please add a example project of opening camera and the taking picture and then crop?

Reply:
Opening the camera to take photos is the same problem. Note that the problem is not `srcFile`, but `corpFile`parent directory does not exist